app/Http/Controllers/Webapp/PostController.php

```diff
@@ -0,0 +1,1890 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: macintosh
+ * Date: 6/8/16
+ * Time: 4:41 PM
+ */
+
+namespace App\Http\Controllers\Webapp;
+
+
+use App\Exceptions\TitleDuplicatedException;
+use App\Models\Activity;
+use App\Models\Backup;
+use App\Models\Keywords;
+use App\Libs\CdsApiDriver;
+use App\Libs\MessagesContainer\Message;
+use App\Libs\MessagesContainer\MessageSet;
+use App\Models\Category;
+use App\Models\Post;
+use App\Models\PostHistory;
+use App\Models\PostMeta;
+use App\Models\Site;
+use App\Models\User;
+use Carbon\Carbon;
+use Illuminate\Http\Request;
+use Illuminate\Routing\Route;
+use Illuminate\Support\Facades\DB;
+use Nayjest\Grids\Components\FiltersRow;
+use Nayjest\Grids\Components\RenderFunc;
+use Nayjest\Grids\Components\SelectFilter;
+use Nayjest\Grids\EloquentDataProvider;
+use Nayjest\Grids\FilterConfig;
+use Nayjest\Grids\GridConfig;
+use Nayjest\Grids\DbalDataProvider;
+use Nayjest\Grids\FieldConfig;
+use Nayjest\Grids\Grid;
+use Nayjest\Grids\ObjectDataRow;
+use Nayjest\Grids\Components\THead;
+use Nayjest\Grids\Components\TFoot;
+use Nayjest\Grids\Components\RecordsPerPage;
+use Nayjest\Grids\Components\ShowingRecords;
+use Nayjest\Grids\Components\OneCellRow;
+use Nayjest\Grids\Components\ColumnHeadersRow;
+use Nayjest\Grids\Components\Base\RenderableRegistry;
+use Nayjest\Grids\Components\HtmlTag;
+use Nayjest\Grids\Components\Laravel5\Pager;
+
+use HTML;
+use Nayjest\Grids\SelectFilterConfig;
+use Psy\Exception\BreakException;
+
+class PostController extends NeedAuthController
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Class "PostController" has 27 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.relipa.blog/coding_rules#rule_key=php%3AS1448)